### PR TITLE
Add some more checks that io.Copy copies as much as expected

### PR DESCRIPTION
**Pull request by Frassle**
More error checks for https://github.com/pulumi/pulumi/issues/9789

### CODE REVIEW

File: sdk/go/common/resource/asset.go [hunk 1]
```diff
@@ -940,7 +943,7 @@ func (a *Archive) Archive(format ArchiveFormat, w io.Writer) error {
 		if err != nil {
 			return err
 		}
-		_, err := io.Copy(w, ss)
+		_, err = io.Copy(w, ss)
```
Comment (bajtos) [inline sdk/go/common/resource/asset.go:946]: Missing size check?

```suggestion
		n, err = io.Copy(w, ss)
```

Comment (bajtos) [reply sdk/go/common/resource/asset.go:946]: (I am not sure if it's possible to find out what is the size of an `ArchiveFormat` though.)

Comment (Frassle) [reply sdk/go/common/resource/asset.go:946]: Yes, I think I started changing this then realized that there was no expected size to check against.

**Review by iwahbe — APPROVED** [review]
LGTM

**Pull request merged**